resteasy-client/src/main/java/org/jboss/resteasy/plugins/providers/sse/client/SseEventSourceImpl.java

```diff
@@ -291,7 +294,15 @@ private void internalClose()
             });
          }
       }
-      executor.shutdownNow();
+
+      // showdownNow only the objects registered by this class
+      for(Future<?> f : futureEventHandlerList) {
+         if (!f.isDone()) {
+            f.cancel(true);
```

Comment:
This looks good to me. Just two minor things: 
From Future javadoc, looks f.cancel() checks if the task is already completed. Can we remove f.isDone() check?
After cancel the future,  do we still need executor.awaitTermination(timeout, unit)?

Reply:
I've removed f.isDone.
We don't want to call executor.awaitTermination(timeout, unit).  If the user is using executor for other tasks we do not want canceling the EventHandler task to terminate all other task.  The user's code should call executor.awaitTermination(timeout, unit) before exiting his app.